### PR TITLE
Better Image Removal

### PR DESCRIPTION
**Pull request by ric-evans**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -34,9 +32,24 @@ inputs:
...
+        # CREATION MODE
+        elif [[ -z "${{ inputs.docker_tags }}" ]]; then
+          echo "Input Error: 'docker_tags' is empty string"
+          exit 1
+        fi
+
     - uses: actions/setup-python@v4
       with:
-        python-version: '3.10'  # NOTE: update, someday
+        python-version: '3.11'  # NOTE: update, someday
```
Comment (dsschult) [inline action.yml:52]: You can probably remove the comment now

File: request_removal.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        ]
+        # remove all matches from lines
+        lines = [ln for ln in lines if ln not in matches]
+        # if an image was negated, strip that & add, but only if it's unique (retain original order)
+        cvmfs_image_str = [s := m.lstrip("-") for m in matches if s not in matches]
+
+    # append negations of each
+    for img in cvmfs_image_str:
+        lines.append(f"-{img}")
+        logging.debug(f"Added line to {DOCKER_IMAGES_FILE}: {lines[-1]}")
```
Comment (dsschult) [inline request_removal.py]: There seems to be a lot of iterations going on here?  Wouldn't it be simpler to do:

```python
    # read
    with open(DOCKER_IMAGES_FILE, "r") as f:
        lines = [ln.strip() for ln in f.readlines()]  # rm each trailing '\n'
    
    matcher = re.compile(rf"^[^-].+ {args.remove_regex_path}$", ln)  # only compile regex once
    lines = ["-"+ln if matcher.match(ln) else ln for ln in lines]  # negate matched lines
```

**Review by dsschult — CHANGES_REQUESTED** [review]

File: action.yml (hunk 1 above)
Comment (ric-evans) [reply action.yml:52]: someday is 3.12 now :smile:

File: request_removal.py (hunk 2 above)
Comment (ric-evans) [reply request_removal.py]: The only scenario I see this not working is one where were have `-A` in the file but no `A`. Then again, that's not a problem... :laughing:

**Review by dsschult — APPROVED** [review]

**Pull request merged**